joomla/joomla-cms
# [5.1] SEF: Implementing index.php behavior

### Summary of Changes
Joomla has been improving its SEO performance constantly and one issue which is still open is the behavior of `index.php` in URLs. In some rare cases, the URL might end with `index.php`, even though that is not needed anymore, resulting in duplicate content, since the same URL without `index.php` gets the same result. A common case is the URL for the homepage, which often enough contains that reference. Another case is sites which switch from non-rewrite to rewrite the URLs.

This PR introduces one new setting in the SEF system plugin. This setting enforces removing the index.php from the URL with a redirect when the URL starts or ends with index.php. This setting only takes effect if URL rewriting is enabled. 

This PR depends on #42692.

I'd like to thank [ithelps Digital](https://www.ithelps-digital.com/) for sponsoring this feature.

### Testing Instructions
1. Apply the PR
2. Enable the option in the SEF system plugin.
3. Open a URL with /index.php/ at the beginning and see that the site is redirected to the version without.
6. Open a URL with /index.php at the end and see that the site is redirected to the version without.


### Link to documentations
Please select:
- [X] Documentation link for docs.joomla.org: https://docs.joomla.org/Search_Engine_Friendly_URLs
